vector/src/main/java/im/vector/VectorApp.java

```diff
@@ -393,6 +394,15 @@ private void stopActivityTransitionTimer() {
 
         mIsCallingInBackground = false;
         mIsInBackground = false;
+        setTheme();
+    }
+
+    private void setTheme() {
```

Comment:
I think you could do that in your `ThemeUtils` class. I.e, maybe the first time you try to apply the theme to an activity, if `currentTheme`  is null, you could then retrieve the value in SharedPreference (or give it a default value).